### PR TITLE
Basic implementation

### PR DESCRIPTION
**Pull request by xy1xy**
This is the implementation without using streams.
I've included an integration test, to prove it works

### CODE REVIEW

**Comment by nblumhardt** [comment]
Great! I should be able to take a proper look through this later today 👍

File: src/index.ts [hunk 1]
```diff
@@ -1,54 +1,85 @@
...
+      this.logger.flush().then(() => {
+        setImmediate(() => this.emit('flushed', this.logger))
+        this.logger.close().then(() => {
+          setImmediate(() => this.emit('closed', this.logger))
+        })
+      })
+    })
+  }
+
+  private mapLevel (level: string) {
```
Comment (nblumhardt) [inline src/index.ts:69]: There may not be any need for this anymore - recent Seq versions work reasonably with arbitrary level names, so it's preferable to pass these through. The `silly` and `http` levels will show as if they're `Information`, but that seems reasonable enough (and is more of a server-side problem than something to handle client-side, now).

File: src/index.ts [hunk 2]
```diff
@@ -1,54 +1,85 @@
...
-      //   level: 'Information',
-      //   messageTemplate: 'Hello for the {n}th time, {user}!',
-      //   // exception: ,
-      //   properties: {
-      //     user: process.env.USERNAME,
-      //     nn: 20,
-      //   },
-      // })
+      this.logger.emit({
+        timestamp: new Date(),
```
Comment (nblumhardt) [inline src/index.ts:45]: Perhaps the `new Date()` here could be lifted out into a variable captured before `setImmediate()`? If it's within the `setImmediate()` callback, other logging statements called after the current call to `info()` could still get their callbacks scheduled before this one, leaving the timestamps out-of-order.

Comment (xy1xy) [reply src/index.ts:45]: Good point - I'll fix this up.

File: src/index.ts (hunk 1 above)
Comment (xy1xy) [reply src/index.ts:69]: OK - I'll give it a try tonight and see how it goes

Comment (xy1xy) [reply src/index.ts:69]: Done :zap:

File: src/index.ts (hunk 2 above)
Comment (xy1xy) [reply src/index.ts:45]: done :zap:

File: src/index.ts (hunk 1 above)
Comment (nblumhardt) [reply src/index.ts:69]: 😎 

Doesn't look like the changeset with these changes has been pushed yet?

Comment (xy1xy) [reply src/index.ts:69]: Sorry - this didn't work - reverted
About to push now

**Comment by nblumhardt** [comment]
Ah - I see -

https://github.com/datalust/seq-logging/blob/dev/seq_logger.js#L153

I'll PR a change into `seq-logging` to remove the level name restriction 👍

**Review by nblumhardt — APPROVED** [review]

**Comment by xy1xy** [comment]
> Ah - I see -
> 
> https://github.com/datalust/seq-logging/blob/dev/seq_logger.js#L153
> 
> I'll PR a change into `seq-logging` to remove the level name restriction +1

Oh! I understand now too - seq-logging will just do a pass-through for level, then the seq server will do its best to parse/match.
Thanks :+1:

**Pull request merged**